src/ChatPage.cpp

```diff
@@ -324,9 +326,27 @@ ChatPage::ChatPage(QSharedPointer<UserSettings> userSettings, QWidget *parent)
                   }
 
                   QSize dimensions;
-                  if (mimeClass == "image")
+                  QString blurhash;
+                  if (mimeClass == "image") {
                           dimensions = QImageReader(dev.data()).size();
 
+                          QImage img;
+                          img.loadFromData(bin);
+                          if (img.height() > 200 && img.width() > 360)
```

Comment:
why height and width greater than 200 & 360 ? is this a minimum requirement for blurhash or something? trying to reduce processing & memory usage for below?

Reply:
would a spanner get thrown if someone uploaded an image that was 2000px h x 360w?

Reply:
No, you can generate a blurhash of pretty much anything. Doing 50000000000x5000000000 is just a bit slow though and doesn't give you a nicer image, since you shring it down to 20 bytes anyway. So either the encoder downscales or you downscale before encoding. Since Qt has that probably a lot more optimized, I'm using their scaling. I also keep the aspect ratio, so no banner or other weirdness should exist. I could probably do a square check though, currently I just use the default values that every image on blurha.sh is. 500000000000000000000x100 pixels could be slow with the code above though, but I think it isn't very common...